lib/submissions.ts

```diff
@@ -0,0 +1,73 @@
+import axios from 'axios';
+import { Submission, StepAnswers } from 'data/flexibleForms/forms.types';
+
+const { ENDPOINT_API, AWS_KEY } = process.env;
+
+const headersWithKey = {
+  'x-api-key': AWS_KEY,
+};
+
+export const startSubmission = async (
+  formId: string,
+  socialCareId: number,
+  createdBy: string
+): Promise<Submission> => {
+  const { data } = await axios.post(
+    `${ENDPOINT_API}/submissions`,
+    {
+      formId,
+      socialCareId,
+      createdBy,
+    },
+    {
+      headers: headersWithKey,
+    }
+  );
+  return data;
+};
+
+export const getSubmissionById = async (
+  submissionId: string
+): Promise<Submission> => {
+  const { data } = await axios.get(
+    `${ENDPOINT_API}/submissions/${submissionId}`,
+    {
+      headers: headersWithKey,
+    }
+  );
+  return data;
+};
+
+export const patchSubmissionForStep = async (
+  submissionId: string,
+  stepId: string,
+  editedBy: string,
+  stepAnswers: StepAnswers
+): Promise<Submission> => {
+  const { data } = await axios.patch(
+    `${ENDPOINT_API}/submissions/${submissionId}/steps/${stepId}`,
+    {
+      stepAnswers,
+      editedBy,
+    },
+    {
+      headers: headersWithKey,
+    }
+  );
+  return data;
+};
+
+export const finishSubmission = async (
+  submissionId: string
+): Promise<Submission> => {
+  const { data } = await axios.patch(
+    `${ENDPOINT_API}/submissions/${submissionId}`,
+    {
+      submittedAt: new Date(),
```

Comment:
When we finish a submission will the submittedAt ever be dynamic.

I'd actually be quite keen to remove this from the JSON body and allow the backend to get DateTime.Now.

Reply:
Also should we provide the value for the worker who is doing the submission, do we want to store SubmittedBy in the backend?

Reply:
it will always be the current time, yes

Reply:
it's definitely important to distinguish between the creator and editors, but i'm not sure whether it matters much who hits the "finish" button. let's leave it as is for now and enhance it later if we need.